### PR TITLE
Add CONFIGs for Mellanox Bluefield SOC in ARM64

### PR DESCRIPTION
**Pull request by jcamposeco**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

This PR adds configurations needed for Mellanox SmartNics, includes drivers needed for their ARM64 SOC. The new size of the RPM  kernel rpm is 3,286,056,960

###### Change Log  

- Added the following CONFIGs
CONFIG_MELLANOX_PLATFORM=y
CONFIG_MLXREG_HOTPLUG=m
CONFIG_MLXREG_IO=m
CONFIG_MLXBF_TMFIFO=m
CONFIG_MLXBF_BOOTCTL=m
CONFIG_MLXBF_PMC=m
CONFIG_MLXBF_GIGE=m
CONFIG_MMC_DW=m
CONFIG_MMC_DW_PLTFM=m
CONFIG_MMC_DW_BLUEFIELD=m
CONFIG_I2C_MLXBF=m
CONFIG_GPIO_MLXBF2=m
CONFIG_POWER_MLXBF=m

- Modified the required config scripts

###### Does this affect the toolchain?  

NO

###### Associated issues  

- NA

###### Links to CVEs  
- NA

###### Test Methodology

- Pipeline build id: BuddyBuild https://dev.azure.com/mariner-org/mariner/_build/results?buildId=390223&view=results
- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=390226&view=results

### CODE REVIEW

File: SPECS/kernel/kernel.spec [hunk 1]
```diff
@@ -186,7 +186,7 @@ if [ -s config_diff ]; then
     echo "Update config file to set changed values explicitly"
 
 #  (DISABLE THIS IF INTENTIONALLY UPDATING THE CONFIG FILE)
-    exit 1
+#    exit 1
```
Comment (anphel31) [inline SPECS/kernel/kernel.spec:189]: please revert this change

File: toolkit/scripts/mariner-required-configs.json [hunk 2]
```diff
@@ -824,6 +824,162 @@
...
+            "CONFIG_MELLANOX_PLATFORM": {
+                "value": [
+                    "y"
+                ],
+                "arch": [
+                    "ARM64"
+                ],
+                "comment": "Needed for Mellanox DPUs",
+                "PR": [
+                    "https://github.com/microsoft/CBL-Mariner/pull/5788"
```
Comment (anphel31) [inline toolkit/scripts/mariner-required-configs.json:837]: I think this should be https://github.com/microsoft/CBL-Mariner/pull/5789?

**Review by anphel31 — APPROVED** [review]

File: SPECS/kernel/kernel.spec (hunk 1 above)
Comment (jcamposeco) [reply SPECS/kernel/kernel.spec:189]: will revert now

File: toolkit/scripts/mariner-required-configs.json (hunk 2 above)
Comment (jcamposeco) [reply toolkit/scripts/mariner-required-configs.json:837]: Right, I add the review I created before, will fix.

**Review by rlmenge — COMMENTED** [review]
Changes look ok to me. Before I approve, can you please boot this kernel and indicate the boot time results and the size of the package similar to https://github.com/microsoft/CBL-Mariner/pull/5657

**Review by rlmenge — APPROVED** [review]
